src/main/java/uk/gov/hmcts/divorce/systemupdate/service/task/RegenerateConditionalOrderPronouncedCoverLetter.java

```diff
@@ -25,54 +25,60 @@ public CaseDetails<CaseData, State> apply(CaseDetails<CaseData, State> caseDetai
+    static void removeAndRegenerateApplicant2(Long caseId, CaseData caseData,
+                                              ConditionalOrderPronouncedCoverLetterHelper coverLetterHelper) {
+        boolean anyDocRemoved = removeExistingCoverLetterIfAny(caseData, CONDITIONAL_ORDER_GRANTED_COVERSHEET_APP_2);
 
-            if (anyDocRemoved) {
-                log.info("Regenerating applicant 1 conditional order pronounced coversheet for case id {} ", caseId);
+        if (anyDocRemoved) {
+            if (caseData.getApplicationType().isSole()) {
+                log.info("Generating respondent conditional order pronounced coversheet for case id {} ", caseId);
+                coverLetterHelper.generateConditionalOrderPronouncedCoversheetOfflineRespondent(
+                    caseData,
+                    caseId,
+                    caseData.getApplicant2(),
+                    caseData.getApplicant1()
+                );
+            } else {
+                log.info("Generating applicant 2 conditional order pronounced coversheet for case id {} ", caseId);
                 coverLetterHelper.generateConditionalOrderPronouncedCoversheet(
                     caseData,
                     caseId,
-                    caseData.getApplicant1(),
-                    CONDITIONAL_ORDER_GRANTED_COVERSHEET_APP_1
+                    caseData.getApplicant2(),
+                    CONDITIONAL_ORDER_GRANTED_COVERSHEET_APP_2
                 );
-
-                caseData.getApplicant1().setCoPronouncedCoverLetterRegenerated(YES);
             }
-        }
 
-        if (caseData.getApplicant2().isApplicantOffline() && caseData.getApplicant2().isConfidentialContactDetails()) {
-            boolean anyDocRemoved = removeExistingCoverLetterIfAny(caseData, CONDITIONAL_ORDER_GRANTED_COVERSHEET_APP_2);
-
-            if (anyDocRemoved) {
-                if (caseData.getApplicationType().isSole()) {
-                    log.info("Generating respondent conditional order pronounced coversheet for case id {} ", caseId);
-                    coverLetterHelper.generateConditionalOrderPronouncedCoversheetOfflineRespondent(
-                        caseData,
-                        caseId,
-                        caseData.getApplicant2(),
-                        caseData.getApplicant1()
-                    );
-                } else {
-                    log.info("Generating applicant 2 conditional order pronounced coversheet for case id {} ", caseId);
-                    coverLetterHelper.generateConditionalOrderPronouncedCoversheet(
-                        caseData,
-                        caseId,
-                        caseData.getApplicant2(),
-                        CONDITIONAL_ORDER_GRANTED_COVERSHEET_APP_2
-                    );
-                }
-
-                caseData.getApplicant2().setCoPronouncedCoverLetterRegenerated(YES);
-            }
+            caseData.getApplicant2().setCoPronouncedCoverLetterRegenerated(YES);
```

Comment:
Do we need to make this flag match the notification flag used [in the query](https://github.com/hmcts/nfdiv-case-api/pull/3624/files#diff-b204569affa343ecc41a9ac3342cf50199f991bbd4eea5b1f1dd42279cb35936R32)? 🤔

Reply:
this class is the original, I've just needed to refactor it a bit because sonar didn't like the duplication of code. That flag is the one checked later to see if something actually needs resending I think